Fix Wy matrix in residual calculation

dmcdougall opened the pull request:
Please merge #635 before merging this PR.

dmcdougall committed:
Wy is a precision matrix, not and obs error matrix

So we call inverse() on it.  There's definitely a better way to do this.
Let's see if Roy will ask me to do the better way instead of the gross way.

roystgnr commented on src/gp/src/GPMSA.C:
```diff
@@ -1235,7 +1235,7 @@ GPMSAFactory<V, M>::setUpHyperpriors()
       }
 
       M& B = *m_BMatrix;
-      M& Wy = *m_observationErrorMatrix;
```
Weren't we saving both the precision and the error forms of this matrix now?

dmcdougall committed:
Fix bug in K setup

We're taking eigenvectors from the wrong place.  We now take them from the
truncated SVD list, as they're in the correct order.

roystgnr commented:
We'll need to cherry pick it into a new branch, of course, and we need to go back later to implement this in a better way, but otherwise the second commit here looks fine.

roystgnr commented:
Actually, wait.  How did that second commit only affect the test_gpmsa_autoscaled_samples.m regression solution, and not test_gpmsa_vector_samples.m too?

roystgnr commented:
Closing in favor of #643, which has a more efficient version of the Wy fix and isn't based off a broken PR.
